collectors/cronjob.go

```diff
@@ -27,7 +27,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 
-	v2batch "k8s.io/api/batch/v2alpha1"
+	v2batch "k8s.io/api/batch/v1beta1"
```

Comment:
@f0 Seems that `batch/v2alpha1` is still available for k8s 1.8

Reply:
@brancz This is where we need to make resource group version configurable. I.e., for a single resource, users can specify which resource version to use aside from the default one which is for now we use the longest support one.

Reply:
I have no idea about whether it is possible to implement this. Because we need to change which package to import for the user specified group version. 

Automatically group version negotiation is also an option. But it is out of control when the auto detect group version is not the desired one. 

What i thought is to support one command option to let user to specify the desired resource group version.

Reply:
I think making this generally configurable is going to create chaos. I believe we should just stick to the oldest API available in the latest Kubernetes release. That way we have the broadest compatibility matrix, while keeping it simple.

Reply:
@andyxning yes, but its recomended to move to v1beta1. v2alpha1 needs to be enabled explicit, v1beta1 is enabled per default (because it's beta now).

Reply:
I would expect that once the beta API is automatically enabled the alpha API is still available and also autoenabled for backward compatibility.

Reply:
@brancz  yes its available, but if i read the changelog right, there are some downsides with v2alpha1
From the Changelog:
'''
Be aware that if you specify the deprecated version, you may encounter Resource Not Found errors. These errors occur because the new controllers look for the new version during a rolling update.
'''

Reply:
I see. Then we should move to v1beta1 I think. Generally we can't give any guarantees around alpha objects I think. @andyxning we should note that in the compatibility guarantees.